open-api/rest-catalog-open-api.yaml

```diff
@@ -2893,6 +3003,37 @@ components:
           additionalProperties:
             type: string
 
+    AppendDataFileRequest:
+      type: object
+      required:
+        - data-files
+      properties:
+        identifier:
+          allOf:
+            - $ref: '#/components/schemas/TableIdentifier'
+          description: Table identifier to update; must be present for CommitTransactionRequest
+        accept-delay-ms:
+          type: integer
+          format: int64
+          description: Acceptable delay in milliseconds for processing the append request.
+          example: 300000
+        data-files:
+          type: array
+          items:
+            $ref: '#/components/schemas/DataFile'
+          description: A list of serialized data files to append to the table metadata.
+
+    AppendDataFileResponse:
+      type:
+        object
+      properties:
+        location:
```

Comment:
I feel like this should be some sort of ID instead of a URL/location?

Reply:
We could technically make the spec more minimal by not even having this but then clients won't have a way to take action based on their expectations of commit time which seems to cross the line imo into not being a useful spec change.

Note, if we do have an ID that's returned and if we establish we do want clients to be able to take action based on their expectations if the operation committed or not, within some time period, I think we'd need to have a separate endpoint for fetching the status of the append. This does expand the spec a bit but again, feel like without it it's not useful to clients. cc @danielcweeks @jackye1995 @rdblue @dramaticlly  thoughts?